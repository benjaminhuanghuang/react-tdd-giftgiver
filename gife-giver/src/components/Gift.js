import React, { Component } from 'react'
import { Form, Button } from 'react-bootstrap'

export default class Gift extends Component {
  constructor(params) {
    super();
    this.state = {
      person: '',
      present: ''
    }
  }

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value })
  }

  render() {
    return (
      <div className='gift'>
        <Form>
          <Form.Group>
            <Form.Label>Person</Form.Label>
            <Form.Control id="person" className='input-person' onChange={this.handleChange}></Form.Control>
          </Form.Group>
          <Form.Group>
            <Form.Label>Present</Form.Label>
            <Form.Control id="present" className='input-present' onChange={this.handleChange}></Form.Control>
          </Form.Group>
        </Form>
        <Button className="btn-remove" onClick={()=>this.props.removeGift(this.props.gift.id)}>
          Remove Gift
        </Button>
      </div>
    )
  }
}
