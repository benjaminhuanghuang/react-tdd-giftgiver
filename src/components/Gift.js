import React, { Component } from 'react'
import { Form, FormGroup, FormControl, ControlLabel, Button } from 'react-bootstrap'

export default class Gift extends Component {
  constructor(params) {
    super();
    this.state = {
      person: '',
      present: ''
    }
  }
  render() {
    return (
      <div>
        <Form>
          <FormGroup>
            <ControlLabel>Person</ControlLabel>
            <FormControl className='input-person' onChange={event=>this.setState({person:event.target.value})}></FormControl>
          </FormGroup>
        </Form>
      </div>
    )
  }
}
 