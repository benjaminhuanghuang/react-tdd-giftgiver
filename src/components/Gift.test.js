import React from "react"
import { shallow } from 'enzyme'
//
import Gift from './Gift'
describe('Gift', () => {
  const gift = shallow(<Gift />);

  it("renders correnctly", () => {
    expect(gift).toMatchSnapshot()
  })


  it("init a persion and present in `state`", () => {
    expect(gift.state()).toEqual({
      person:'',
      present:''
    })
  })


  describe('When typing into the person instpu', ()=>{
    beforeEach(()=>{
      gift.find('.input-person').simulate('change', {target:{value:'abcd'}})
    })

    afterEach(()=>{
      gift.setState({gifts:[]})
    })
    
    it("updates the person in state", () => {
      expect(gift.state().person).toEqual('abcd')
    })
  
  })

})




