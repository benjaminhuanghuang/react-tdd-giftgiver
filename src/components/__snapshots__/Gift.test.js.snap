// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correnctly 1`] = `ShallowWrapper {}`;

exports[`Gift init a persion and present in \`state\` 1`] = `ShallowWrapper {}`;

exports[`Gift renders correnctly 1`] = `ShallowWrapper {}`;
